sql/src/main/java/org/apache/druid/sql/calcite/table/DruidTable.java

```diff
@@ -106,7 +106,10 @@ public Statistic getStatistic()
   @Override
   public RelDataType getRowType(final RelDataTypeFactory typeFactory)
   {
-    return RowSignatures.toRelDataType(rowSignature, typeFactory);
+    // For external datasources, the row type should be determined by whatever the row signature has been explicitly
+    // passed in. Typecasting directly to SqlTypeName.TIMESTAMP will lead to inconsistencies with the Calcite functions
```

Comment:
I think that for future reader, it will be good to clarify with an example of inconsistency.

Reply:
are these changes still needed?

Reply:
Yes, I think so. This is because this change is required for the Calcite layer. The two use cases that I am thinking of rn:
1. Let's say we have a function FUNC(a) that accepts long. If the user passes a `__time` column of type long, this would be typecasted to SqlTypeName.TIMESTAMP before passing to FUNC(a). This will cause a signature mismatch, even though the query would have produced correct results downstream. 
2. In the future, when the typecasting with the cursor is improved to accept columns of type other than LONG, this would ensure that the explicit type passed by the user is respected.